src/shared/acl-util.c

```diff
@@ -209,26 +209,50 @@ int acl_search_groups(const char *path, char ***ret_groups) {
         return ret;
 }
 
-int parse_acl(const char *text, acl_t *ret_acl_access, acl_t *ret_acl_default, bool want_mask) {
-        _cleanup_free_ char **a = NULL, **d = NULL; /* strings are not freed */
+int parse_acl(
+                const char *text,
+                acl_t *ret_acl_access,
+                acl_t *ret_acl_access_cond_exec,
```

Comment:
maybe name this just `ret_acl_access_exec` or `ret_acl_access_xbit` or so

and add a quick comment to the line indicating briefly what this is. i.e. `/* extra rules to apply to inodes subject to uppercase X handling, i.e. already marked executable */`